main.role_assignments.tf

```
@@ -9,3 +9,8 @@ data "azapi_resource_list" "role_definitions" {
     results = "value[].{id: id, role_name: properties.roleName}"
   }
 }
+
+#  Use a random UUID for the role assignment name if the variable is set to true.
+resource "random_uuid" "role_assignment_name" {
+  for_each = var.role_assignment_name_use_random_uuid ? var.role_assignments : {}
+}
```

```suggestion

  keepers = {
    principal_id = each.principal_id
    role_definition_id_or_name = each.role_definition_id_or_name
  }
}
```

Maybe additionally scope should be considered:
```suggestion

  keepers = {
    principal_id = each.principal_id
    role_definition_id_or_name = each.role_definition_id_or_name
    scope_id = var.role_assignment_definition_scope
  }
}
```